### PR TITLE
Nonce simplification

### PR DESCRIPTION
**Pull request by expede**
* Remove `Nonce12` and `generate_12`
* `Nonce16` generation longer takes a `seed`

### CODE REVIEW

File: src/crypto/nonce.rs [hunk 1]
```diff
@@ -34,30 +31,15 @@ pub enum Nonce {
...
-                } else {
-                    false
-                }
-            }
             (Nonce::Custom(a), Nonce::Nonce16(b)) => {
                 if a.len() == 16 {
                     a.as_slice() == b
                 } else {
                     false
                 }
```
Comment (matheus23) [inline src/crypto/nonce.rs]: Unrelated to this PR's changes, but just caught my eye: This expression can probably be replaced with `a.as_slice() == b` in its entirety. In any case - this comment doesn't block this from being merged.

File: src/crypto/nonce.rs [hunk 2]
```diff
@@ -158,18 +91,16 @@ impl Nonce {
...
-
-        let buf = salt.as_mut_slice();
-        getrandom(buf).expect("irrecoverable getrandom failure");
+    pub fn generate_16() -> Nonce {
+        let mut buf = [0; 16];
+        getrandom(&mut buf).expect("irrecoverable getrandom failure");
 
         let mut hasher = Sha2_256::default();
-        hasher.update(buf);
+        hasher.update(&mut buf);
```
Comment (matheus23) [inline src/crypto/nonce.rs]: Hm. I don't understand why we should hash the output of `getrandom`?
I think it's acceptable to assume `getrandom` gives us good quality randomness, but even if it didn't the SHA256 wouldn't make the nonces more unique.

File: src/crypto/nonce.rs (hunk 1 above)
Comment (expede) [reply src/crypto/nonce.rs]: Oh yeah, good point since this is now the only case 👍 That's quick; fixing now

File: src/crypto/nonce.rs (hunk 2 above)
Comment (expede) [reply src/crypto/nonce.rs]: Ah, yeah indeed we don't need a bunch of this other stuff, including the resizing below 👍

**Review by matheus23 — APPROVED** [review]
Looking great! :) Thanks!

**Pull request merged**